File: protocol-generator/java/src/main/java/org/entitycore/protocol/codec/NonCanonicalEcfException.java
package org.entitycore.protocol.codec;

/** Wire bytes are not canonical ECF (indefinite length, non-minimal arg, trailing
 *  bytes, reserved additional-info, max-depth, bad simple value). Status 400. */
public class NonCanonicalEcfException extends EntityCodecException {
    private static final long serialVersionUID = 1L;
    public NonCanonicalEcfException(String message) { super(message); }
}

File: protocol-generator/php/src/NonCanonicalEcfException.php
<?php

declare(strict_types=1);

namespace EntityCore;

/**
 * A wire input the canonical decoder MUST reject: a CBOR tag (major type 6,
 * invariant N2), an indefinite length, a non-minimal argument, a reserved
 * additional-info value, duplicate map keys, trailing bytes, or over-depth.
 *
 * The peer maps this to 400 non_canonical_ecf at the dispatch boundary.
 */
class NonCanonicalEcfException extends CodecException
{
}
